### ISSUE TITLE
Migrate from Mercurial

### ISSUE THREAD

**Issue by thomaskeller79** [prost-planner/prost#76]
**[Original report](https://bitbucket.org/tkeller/prost/issue/76) by geisserf (Bitbucket: [557058:e7a9f9a5-3ea8-4154-97d2-10446425dce3](https://bitbucket.org/557058:e7a9f9a5-3ea8-4154-97d2-10446425dce3), GitHub: [geisserf](https://github.com/geisserf)).**

----------------------------------------

Since Bitbucket ceases to support Mercurial next year we should think about migrating from Mercurial to another CVS. I vote for git, but I am happy to discuss other options.

**Comment by thomaskeller79** [comment]
I have migrated the master repo as well as the issue tracker and the wiki (without adjusting the latter) here and have added a comment to the wiki at bitbucket that the prost repo now lives here.

The following is still missing:
- add description how to migrate a mercurial Prost repo to a git repo that is compatible with the new master repo
- update the wiki
- migrate the prost-dev repo

**Comment by thomaskeller79** [comment]
All of the points in my previous comment have been implemented. From my side, we can close this issue. Please let me know if anything is still missing.

**Comment by geisserf** [comment]
The TODO in the [development workflow](https://github.com/prost-planner/prost/wiki/Workflow) is not yet done! If you want I can take care of that too.

**Comment by thomaskeller79** [comment]
Yes, I guess that should be part of this issue... It would be great if you would take care of this!

**Comment by geisserf** [comment]
I updated the wiki entry. Some links to other wiki sites are broken, I fix this when I add the wiki of the unit test tutorial. I did not test point 14 of the workflow, if the command is different in the end we have to update the wiki.

Otherwise this can be closed from my side.

**Comment by geisserf** [comment]
I think something is not set up correctly in the prost-dev repo. When I pull from the master branch, everything is up to date. But once I switch to a new branch and want to push I get the message that my branch is behind.

After applying `git branch --set-upstream-to=origin/issue-62 issue-62` and pulling again I saw that the testbed/experiment.py script has a conflict.

**Comment by geisserf** [comment]
> I updated the wiki entry. Some links to other wiki sites are broken, I fix this when I add the wiki of the unit test tutorial. I did not test point 14 of the workflow, if the command is different in the end we have to update the wiki.

I added the unit tests tutorial and updated the hyperlinks of the wiki.

**Comment by thomaskeller79** [comment]
> I updated the wiki entry. Some links to other wiki sites are broken, I fix this when I add the wiki of the unit test tutorial. I did not test point 14 of the workflow, if the command is different in the end we have to update the wiki.

Shouldn't the branch name be master?

Also, in point 4, there are double quotes around the message, and in point 3, there is a single quote sign that doesn't have an end.

Thanks for taking care of this!

**Comment by thomaskeller79** [comment]
> Update:
> 
> I did not notice that the prost-dev repo has the same branches as on bitbucket, and instead assumed my issue branch was new (so I worked on a branch which already existed). Nothing wrong then.

Can you have a look at this? It would be good to resolve this soon and I won't find the time this week.

**Comment by geisserf** [comment]
> Shouldn't the branch name be master?

What do you mean?

> Also, in point 4, there are double quotes around the message, and in point 3, there is a single quote sign that doesn't have an end.

Fixed.

> Can you have a look at this? It would be good to resolve this soon and I won't find the time this week.

The problem was that my issue branch was a couple of revisions behind, and some of the files conflicted. There is not really anthing we can do here. For the older issue-x branches it will probably also be easier to delete the branch and create a new one to work on.

**Comment by thomaskeller79** [comment]
> > Shouldn't the branch name be master?
> 
> What do you mean?
> 

OK, I see now that this was  not enough information ;)

In point 15 (I assume that's the one you didn't test yet), the command is `git push --branch default --new-branch https://github.com/prost-planner/prost`, but I believe it should read `git push --branch master --new-branch https://github.com/prost-planner/prost`

> > Also, in point 4, there are double quotes around the message, and in point 3, there is a single quote sign that doesn't have an end.
> 
> Fixed.
> 
> > Can you have a look at this? It would be good to resolve this soon and I won't find the time this week.
> 
> The problem was that my issue branch was a couple of revisions behind, and some of the files conflicted. There is not really anthing we can do here. For the older issue-x branches it will probably also be easier to delete the branch and create a new one to work on.

Does that mean it is fixed already? Also, (I asked this already in your pull request), how is it possible that something vanishes from the history entirely? Did you strip the old issue-62 branch from the github repo?

**Comment by geisserf** [comment]
> In point 15 

Ah, yes you are right. Fixed (at least if the command is right).

> Does that mean it is fixed already?

Yes, after all there was nothing wrong.

> Did you strip the old issue-62 branch from the github repo?

No, I only deleted issue-62 with the github web-interface and created a new issue-62 branch where I commited my changes. I am not sure if the problem is that I opened the branch with the same name, or if deleted branches are not shown in general on github.
